platform/mbed_enum_flags.h

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright (c) 2020 ARM Limited. All rights reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ * Licensed under the Apache License, Version 2.0 (the License); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef MBED_SCOPED_ENUM_FLAGS_H
+#define MBED_SCOPED_ENUM_FLAGS_H
+
+#include <type_traits>
+
+#define ENUM_FLAG_OPERATOR(T, OP) \
+inline constexpr T operator OP( T lhs, T rhs) \
+{ \
+    return T (static_cast<std::underlying_type_t< T > >(lhs) OP \
+        static_cast<std::underlying_type_t< T > >(rhs)); \
+}
+
+
+#define ENUM_FLAG_OPERATOR_LHS_REFERNCE(T, OP) \
+inline constexpr T & operator OP (T & a, T b) \
```

Comment:
No space between `&` and variable name. Can probably have one before `operator` if you like.